### PR TITLE
Fire leaf flux update

### PR DESCRIPTION
**Pull request by jkshuman**
### Description:
'cfa' was updated to 'fraction_crown_burned' for readability.
The changes here address Issue 412 and inconsistencies within the fire flux routine for live plants. The fire flux for live plants was updated to remove leaves according to the fraction of the crown scorched by fire. Consumption of branches would burn sapwood and remove stemwood, and will be addressed in a separate issue with separate fluxes. 

The equation was also updated to remove the minimum evaluation, as the fraction_crown_burned term already has a check on it to confirm it is not greater than 1 within the fire routine.

### Collaborators:
Charlie Koven, Ryan Knox, Rosie Fisher

### Expectation of Answer Changes:
removal of the sapwood flux will generate answer changes.

### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [ x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ x] FATES PASS/FAIL regression tests were run
- [ ] If answers were expected to change, evaluation was performed and provided

### Test Results:

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:
CTSM commit e49b84fb4

FATES baseline hash-tag:
 FATES commit 91785e3

Test Output:
Compiles and runs on Hobart. Biomass accumulation through 5 year run with and without active fire for 2 PFTs (TropTree and Grass).

### CODE REVIEW

**Comment by ckoven** [comment]
there are a bunch of untyped reals in the fire code.  when i was trying too figure out the cause of my crashes i went through and changed them all to r8s.  the resulting commit is  https://github.com/ckoven/fates/commit/0333260cc353ebeb89659d08daf804d792866d20 -- you might want to cherry-pick it into this?

**Comment by ckoven** [comment]
there's another issue in the fire code that I had also realized when I was trying to figure stuff out in august, which is that the diagnostic fire mortality rate reported by FATES is effectively meaningless, because it is calculated at the end of the timestep using number densities on cohorts that have just suffered very large mortality during that timestep.  I made some changes in https://github.com/ckoven/fates/commit/b841b8f772b1a5d8c67796ce76faa6f9e25bb0ec and https://github.com/ckoven/fates/commit/13db3f8e0c822d01599b4e8f606139fce50e2c03 to address this. not sure if it makes sense to wrap these into this PR, but since the current code reports really wrong mortality rates, I do think we ought to pull these or similar changes into master at some point in case anyone wants to actually look at the fire mortality rates.

**Comment by jkshuman** [comment]
I pulled these updates to r8 into the fire_leaf_flux PR.

------------------------------------------------------------------------
Jacquelyn Shuman, PhD
Project Scientist
Terrestrial Sciences Section
National Center for Atmospheric Research
PO Box 3000
Boulder, Colorado 80307-3000
USA

jkshuman@ucar.edu
office: +1-303-497-1787

File: fire/SFMainMod.F90 [hunk 1]
```diff
@@ -878,34 +878,34 @@ subroutine  crown_damage ( currentSite )
...
+                   currentCohort%fraction_crown_burned = 0.0_r8
                 else
                    ! Flames part of way up canopy. 
                    ! Equation 17 in Thonicke et al. 2010. 
                    ! flames over bottom of canopy but not over top.
                    if ((currentCohort%hite > 0.0_r8).and.(currentPatch%SH >=  &
                         (currentCohort%hite-currentCohort%hite*EDPftvarcon_inst%crown(currentCohort%pft)))) then 
 
-                           currentCohort%cfa =  (currentPatch%SH-currentCohort%hite*(1- &
+                           currentCohort%fraction_crown_burned =  (currentPatch%SH-currentCohort%hite*(1- &
```
[the remark this answers is not in the document]
Comment (jkshuman) [reply fire/SFMainMod.F90]: done

**Comment by jkshuman** [comment]
I think the fire mortality should be its own PR. There is enough going on there to keep it separate. But yes we should get that into master soon.

**Comment by ckoven** [comment]
@jkshuman I agree that making the fire mortality fixes a separate PR makes sense.  and thanks for fixing the r8s!

**Comment by jkshuman** [comment]
I created a new branch with the fire mortality changes @ckoven added. Reviewing those, and will test. Plan to submit fire mortality as a separate PR.

**Comment by rgknox** [comment]
@jkshuman I updated my testing script for cheyenne, it is the top link in the useful stuff wiki page.

https://github.com/NGEET/fates/wiki/Useful-Stuff

You mentioned you wanted to give it a try, let me know if you have any trouble or would like me to test instead.

**Comment by jkshuman** [comment]
Passed baseline tests on Cheyenne.

**Pull request merged**